core/src/apps/management/recovery_device/word_validity.py

```diff
@@ -0,0 +1,103 @@
+from micropython import const
+
+import storage.recovery
+from trezor.messages import BackupType
+
+from apps.management.recovery_device import recover
+
+if False:
+    from typing import List, Optional
+    from trezor.messages.ResetDevice import EnumTypeBackupType
+
+OK = const(0)
+NOK_IDENTIFIER_MISMATCH = const(1)
+NOK_ALREADY_ADDED = const(2)
+NOK_THRESHOLD_REACHED = const(3)
+
+
+def check(
+    current_index: int,
+    current_word: str,
+    backup_type: Optional[EnumTypeBackupType],
+    previous_words: List[str],
+) -> int:
```

Comment:
I am not sure how to type the return type correctly. Both `Union[OK, NOK..]` and `Literal[OK, NOK..]` seems incorrect.

Also, are the constants declared using the `const` something we prefer? Or rather `object()` as is used somewhere else?

Reply:
wouldn't it be better to have a return type `None` and raise to indicate the result?

Reply:
Hm maybe, I guess there is nothing wrong with returning result using an exception.

Reply:
I have rewritten it. Check it out: https://github.com/trezor/trezor-firmware/pull/774/commits/7f730cb6f9a1635d75409e1a721f9879805c6ba1 . Both ways are fine by me, maybe this is slightly nicer